docs/proposals/scheduledjob.md

```diff
@@ -0,0 +1,163 @@
+1. Be able to create a periodic job, eg. database backup, sending emails.
+
+
+## Motivation
+
+ScheduledJobs are needed for performing all time related actions, namely backups,
+report generation and alike.  Each of these tasks should be allowed to perform
+repeatedly (once a day/month, etc.) or once at a given point in time.
+
+
+## Implementation
+
+ScheduledJob controller relies heavily on the [Job Controller API](job.md)
+for running actual jobs, on top of which it adds information regarding the date
+and time part according to ISO8601 format.
+
+The new `ScheduledJob` object will have the following content:
+
+```go
+// ScheduledJob represents the configuration of a single scheduled job.
+type ScheduledJob struct {
+    TypeMeta
+    ObjectMeta
+
+    // Spec is a structure defining the expected behavior of a job, including the schedule.
+    Spec ScheduledJobSpec
+
+    // Status is a structure describing current status of a job.
+    Status ScheduledJobStatus
+}
+
+// ScheduledJobList is a collection of scheduled jobs.
+type ScheduledJobList struct {
+    TypeMeta
+    ListMeta
+
+    Items []ScheduledJob
+}
+```
+
+`ScheduledJobSpec` structure is defined to contain all the information how the actual
+job execution will look like, including the `JobSpec` from [Job Controller API](job.md)
+and the schedule in ISO8601 format.  This implies that each ScheduledJob execution
+will be created from the JobSpec actual at a point in time when the execution will
+be started.
+
+```go
+// ScheduledJobSpec describes how the job execution will look like and when it will actually run.
+type ScheduledJobSpec struct {
+
+    // Template is the object that describes the job that will be created when
+    // executing a ScheduledJob.
+    Template *JobSpec
+
+    // Selector is a label query over created jobs, sorted and distinguished by CreationTimestamp.
+    Selector map[string]string
+
+    // Schedule contains the schedule in ISO8601 format, eg.
+    // - 2015-07-21T14:00:00Z - represents date and time in UTC
+    // - R/2015-07-21T14:00:00Z/P1D - represents endlessly repeating interval (1 day), starting from given date
+    Schedule string
+
+    // ConcurrencyPolicy specifies how to treat concurrent executions of a Job.
+    ConcurrencyPolicy ConcurrencyPolicy
+
+    // BlockOnFailure suspends scheduling of next job runs after a failed one,
+    // by default it is false.
+    BlockOnFailure bool
+}
+
+// ConcurrencyPolicy describes how the job will be handled.
+// Only one of the following concurrent policies may be specified.
+// If none of the following policies is specified, the default one
+// is AllowConcurrent.
+type ConcurrencyPolicy string
+
+const (
+    // AllowConcurrent allows ScheduledJobs to run concurrently.
+    AllowConcurrent ConcurrencyPolicy = "Allow"
+
+    // SkipNextConcurrent forbids concurrent runs, skipping next run if previous
+    // hasn't finished yet.
+    SkipNextConcurrent ConcurrencyPolicy = "SkipNext"
```

Comment:
This is confusing because with SkipNextConcurrent, the object in the statement "skip the next job" is the job that is already running and with CancelPreviousConcurrent, the object in the statement "skip the previous job" is the job that is up to be run. Side note, with AllowConcurrent the subject in the statement "allow this job" is the job that is up to be run like CancelPreviousConcurrent.